app/controllers/register/CompanyRegistrationNumberBaseController.scala

```diff
@@ -0,0 +1,87 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.register
+
+import config.FrontendAppConfig
+import controllers.Retrievals
+import controllers.actions._
+import forms.CompanyRegistrationNumberFormProvider
+import identifiers.TypedIdentifier
+import identifiers.register.establishers.company.CompanyRegistrationNumberId
+import javax.inject.Inject
+import models.requests.DataRequest
+import models.{CompanyRegistrationNumber, Index, Mode}
+import play.api.data.Form
+import play.api.i18n.{I18nSupport, MessagesApi}
+import play.api.mvc.{Action, AnyContent, Call}
+import play.twirl.api.Html
+import services.UserAnswersService
+import uk.gov.hmrc.play.bootstrap.controller.FrontendController
+import utils._
+
+import scala.concurrent.{ExecutionContext, Future}
+
+abstract class CompanyRegistrationNumberBaseController @Inject()(
+                                                           appConfig: FrontendAppConfig,
+                                                           override val messagesApi: MessagesApi,
+                                                           userAnswersService: UserAnswersService,
+                                                           navigator: Navigator,
+                                                           authenticate: AuthAction,
+                                                           getData: DataRetrievalAction,
+                                                           allowAccess: AllowAccessActionProvider,
+                                                           requireData: DataRequiredAction,
+                                                           formProvider: CompanyRegistrationNumberFormProvider
+                                                         )(implicit val ec: ExecutionContext) extends FrontendController with Retrievals with I18nSupport with Enumerable.Implicits with MapFormats {
+
+  val form = formProvider()
+
+  def addView(mode: Mode, index: Index, srn: Option[String])(implicit request: DataRequest[AnyContent]): Html
+
+  def errorView(mode: Mode, index: Index, srn: Option[String], form: Form[_])(implicit request: DataRequest[AnyContent]): Html
+
+  def updateView(mode: Mode, index: Index, srn: Option[String], value: CompanyRegistrationNumber)(implicit request: DataRequest[AnyContent]): Html
+
+  def id(index: Index): TypedIdentifier[CompanyRegistrationNumber]
+
+  def onPageLoad(mode: Mode, srn: Option[String], index: Index): Action[AnyContent] =
+    (authenticate andThen getData(mode, srn) andThen allowAccess(srn) andThen requireData).async {
+      implicit request =>
+        val redirectResult = request.userAnswers.get(id(index)) match {
+          case None =>
+            Ok(addView(mode, index, srn))
+          case Some(value) =>
```

Comment:
We shouldn't be populating the value if user's previous answer was CRN.No(reason)

Reply:
If the users answer was they didn't have a CRN, they shouldn't be shown the page to update/add it. So the user would never be shown the CompanyRegistrationNumberUpdateController page.